### PR TITLE
Scav selling pad

### PR DESCRIPTION
**Pull request by Baldir-Odensen**
## About The Pull Request

adds the selling pad and the computer to control it to the scavengers to be added to the ship.

## Why It's Good For The Game

it would allow the scavengers to sell stuff and thus make money from their scavenging.

## Changelog
:cl:
add: adds a selling pad to for the scavengers along with an accompanying computer
add: adds a budget card for and budget for the scavengers
/:cl:

### CODE REVIEW

File: code/game/machinery/scav_selling_pad.dm [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+	name = "cargo hold pad"
+	icon = 'icons/obj/telescience.dmi'
+	icon_state = "lpad-idle-o"
+	var/idle_state = "lpad-idle-o"
+	var/warmup_state = "lpad-idle"
+	var/sending_state = "lpad-beam"
+	var/cargo_hold_id
+
+/obj/machinery/scav_selling/multitool_act(mob/living/user, obj/item/multitool/I)
+	if (istype(I))
```
Comment (ike709) [inline code/game/machinery/scav_selling_pad.dm:11]: Is it even possible for this check to fail?

File: code/game/machinery/scav_selling_pad.dm [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+		to_chat(user, "<span class='notice'>You register [src] in [I]s buffer.</span>")
```
Comment (ike709) [inline code/game/machinery/scav_selling_pad.dm]:
```suggestion
		to_chat(user, "<span class='notice'>You register \the [src] in [I]'s buffer.</span>")
```

File: code/game/machinery/scav_selling_pad.dm [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+
+/obj/machinery/computer/scav_selling_control/LateInitialize()
+	. = ..()
+	if(cargo_hold_id)
+		for(var/obj/machinery/scav_selling/P in GLOB.machines)
+			if(P.cargo_hold_id == cargo_hold_id)
+				pad = P
+				return
+	else
+		pad = locate() in range(4,src)
```
Comment (ike709) [inline code/game/machinery/scav_selling_pad.dm]: Gross.

File: code/game/machinery/scav_selling_pad.dm [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+	return INITIALIZE_HINT_LATELOAD
+
+/obj/machinery/computer/scav_selling_control/multitool_act(mob/living/user, obj/item/multitool/I)
+	if (istype(I) && istype(I.buffer,/obj/machinery/scav_selling))
+		to_chat(user, "<span class='notice'>You link [src] with [I.buffer] in [I] buffer.</span>")
+		pad = I.buffer
+		updateDialog()
+		return TRUE
+
+/obj/machinery/computer/scav_selling_control/LateInitialize()
```
Comment (ike709) [inline code/game/machinery/scav_selling_pad.dm:38]: IIRC, even using `LateInitialize()`, you can't guarantee that the maploader will load in the other machine first when loading in a shuttle.

File: code/game/machinery/scav_selling_pad.dm [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+/obj/machinery/scav_selling/multitool_act(mob/living/user, obj/item/multitool/I)
+	if (istype(I))
+		to_chat(user, "<span class='notice'>You register [src] in [I]s buffer.</span>")
+		I.buffer = src
+		return TRUE
+
+/obj/machinery/computer/scav_selling_control
+	name = "cargo hold control terminal"
+	var/status_report = "Idle"
+	var/obj/machinery/scav_selling/pad
```
Comment (ike709) [inline code/game/machinery/scav_selling_pad.dm]: This should use weakrefs.

File: code/game/machinery/scav_selling_pad.dm [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+	. = ..()
+	if(cargo_hold_id)
+		for(var/obj/machinery/scav_selling/P in GLOB.machines)
+			if(P.cargo_hold_id == cargo_hold_id)
+				pad = P
+				return
+	else
+		pad = locate() in range(4,src)
+
+/obj/machinery/computer/scav_selling_control/ui_interact(mob/user)
```
Comment (ike709) [inline code/game/machinery/scav_selling_pad.dm:46]: I would strongly prefer a tgui-next UI.

**Review by ike709 — CHANGES_REQUESTED** [review]
Didn't fully review this, already left you with a fair bit to work on though.

**Comment by Baldir-Odensen** [comment]
Don't really know how to do tgui next I mostly used the pirate selling pad and changed a few things

File: code/game/machinery/scav_selling_pad.dm (hunk 1 above)
Comment (Baldir-Odensen) [reply code/game/machinery/scav_selling_pad.dm:11]: no clue it there on the pirate one so i put it there on the scav one

File: code/game/machinery/scav_selling_pad.dm (hunk 5 above)
Comment (Baldir-Odensen) [reply code/game/machinery/scav_selling_pad.dm]: dont really know how to do those.
im mostly learning this as i work on it

File: code/game/machinery/scav_selling_pad.dm (hunk 4 above)
Comment (Baldir-Odensen) [reply code/game/machinery/scav_selling_pad.dm:38]: ok so they will have to reconect it once in a while. it is based on the code for the pirate pad so if the pirates dont have that problem this one shouldnt have it either.

File: code/game/machinery/scav_selling_pad.dm (hunk 2 above)
Comment (Baldir-Odensen) [reply code/game/machinery/scav_selling_pad.dm]: ok its in the latest commit

File: code/game/machinery/scav_selling_pad.dm (hunk 6 above)
Comment (Baldir-Odensen) [reply code/game/machinery/scav_selling_pad.dm:46]: No clue how to set up tgui next.
The pirate selling computer uses the exact same ui code btw

**Comment by qwertyquerty** [comment]
@Baldir-Odensen ?

**Comment by qwertyquerty** [comment]
@Baldir-Odensen

**Comment by qwertyquerty** [comment]
stale

**Pull request closed without merging**